src/connect.sh

```diff
@@ -13,22 +13,22 @@ OutputFile=$3
 
 EncodedPublicKey=$(echo $PublicKey | jq -R -r @uri)
 
-SessionsCommand="sessions"
+IntervalSeconds=55
+TotalIntervals=11
+RestartSeconds=0.3
 
 echo "agent: connecting"
 
 while true; do
 
-  # This initial command causes the server to immediately send the active session after connection.
-  # The stdin stream actually closes after echo completes, so no further commands can be sent.
-  # That's fine since the agent simply listens for messages after its initial command.
-  # The -n flag below is required to ensure that websocat stays open even after stdin EOF.
-  echo $SessionsCommand | \
-    websocat -n wss://agent.reflect.run/api/agents/${EncodedPublicKey} \
+  # Periodically echo the "sessions" command to improve the chances of the websocket staying alive.
+  # But still reset and reconnect every 10 minutes or so, or if the server drops the connection.
+  ./output.sh $IntervalSeconds $TotalIntervals | \
+    websocat wss://agent.reflect.run/api/agents/${EncodedPublicKey} \
```

Comment:
Is the removal of `-n` intentional?

Reply:
Yup - because we'll complete the `output.sh` script in 10+ minutes and we want that to close the websocat process so we can reset. The net effect of this change is we'll reestablish the websocket every ~10 minutes now and use a heartbeat, rather than a single message and then silence before. If we find the 10 minutes is too aggressive then we can bump the 10 minutes to an hour, for example.